File: Chapter08/Lesson3/Exercise2/Completed/VB/MonitorWebSite/Service1.vb
Imports System.Timers
Imports System.IO
Imports System.Net

Public Class MonitorWebSite

    Private t As Timer = Nothing

    Protected Overrides Sub OnStart(ByVal args() As String)
        t.Start()
    End Sub

    Protected Overrides Sub OnStop()
        t.Stop()
    End Sub

    Protected Overrides Sub OnContinue()
        t.Start()
    End Sub

    Protected Overrides Sub OnPause()
        t.Stop()
    End Sub

    Protected Overrides Sub OnShutdown()
        t.Stop()
    End Sub

    Protected Sub t_Elapsed(ByVal sender As System.Object, ByVal e As System.Timers.ElapsedEventArgs)
        Try
            ' Send the HTTP request
            Dim url As String = "http://www.microsoft.com"
            Dim g As HttpWebRequest = CType(WebRequest.Create(url), HttpWebRequest)
            Dim r As HttpWebResponse = CType(g.GetResponse, HttpWebResponse)

            ' Log the response to a text file
            Dim path As String = AppDomain.CurrentDomain.SetupInformation.ApplicationBase + "log.txt"
            Dim tw As TextWriter = New StreamWriter(path, True)
            tw.WriteLine(DateTime.Now.ToString + " for " + url + ": " + r.StatusCode.ToString)
            tw.Close()

            ' Close the HTTP response
            r.Close()
        Catch ex As Exception
            System.Diagnostics.EventLog.WriteEntry("Application", "Exception: " + ex.Message.ToString)
        End Try
    End Sub
End Class

File: Chapter08/Lesson3/Exercise2/Completed/CS/MonitorWebSite/Service1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Diagnostics;
using System.ServiceProcess;
using System.Text;
using System.Timers;
using System.Net;
using System.IO;

namespace MonitorWebSite
{
    public partial class MonitorWebSite : ServiceBase
    {
        private Timer t = null;
        public MonitorWebSite()
        {
            InitializeComponent();

            t = new Timer(10000);
            t.Elapsed += new ElapsedEventHandler(t_Elapsed);
        }

        void t_Elapsed(object sender, ElapsedEventArgs e)
        {
            try
            {
                // Send the HTTP request
                string url = "http://www.microsoft.com";
                HttpWebRequest g = (HttpWebRequest)WebRequest.Create(url);
                HttpWebResponse r = (HttpWebResponse)g.GetResponse();

                // Log the response to a text file
                string path = AppDomain.CurrentDomain.SetupInformation.ApplicationBase + "log.txt";
                TextWriter tw = new StreamWriter(path, true);
                tw.WriteLine(DateTime.Now.ToString() + " for " + url + ": " + r.StatusCode.ToString());
                tw.Close();

                // Close the HTTP response
                r.Close();
            }
            catch (Exception ex)
            {
                System.Diagnostics.EventLog.WriteEntry("Application", "Exception: " + ex.Message.ToString());
            }
        }

        protected override void OnStart(string[] args)
        {
            t.Start();
        }

        protected override void OnStop()
        {
            t.Stop();
        }

        protected override void OnContinue()
        {
            t.Start();
        }

        protected override void OnPause()
        {
            t.Stop();
        }

        protected override void OnShutdown()
        {
            t.Stop();
        }
    }
}
